### PR TITLE
Moved batch to end of processors pipeline

### PR DESCRIPTION
**Pull request by harshit-splunk**
Addresses https://github.com/signalfx/splunk-otel-collector-chart/issues/455

### CODE REVIEW

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 1]
```diff
@@ -158,8 +158,11 @@ exporters:
 
 service:
   telemetry:
+    logs:
+      level: {{ .Values.logLevel }}
     metrics:
       address: 0.0.0.0:8889
+
```
Comment (jvoravong) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:165]: Nit: Extra new line.

**Comment by harshit-splunk** [comment]
> We have a fluentd.config.logLevel config, so I don't recommend we have a global logLevel config. Maybe in another PR add an example of how to enable debug logging like how our docs describe.

Maybe we can depreciate `fluentd.config.logLevel` and use global logLevel config in both fluentd and otel.

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 2]
```diff
@@ -646,6 +645,7 @@ service:
         {{- if .Values.environment }}
         - resource/add_environment
         {{- end }}
+        - batch
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:648]: Please move it right after filter processor. There is no performance improvement in moving it further

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 3]
```diff
@@ -673,11 +673,11 @@ service:
         {{- end }}
       processors:
         - memory_limiter
-        - batch
         - resource
         {{- if .Values.environment }}
         - resource/add_environment
         {{- end }}
+        - batch
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:680]: No change needed here

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 4]
```diff
@@ -699,12 +699,12 @@ service:
       processors:
         - memory_limiter
         - k8sattributes
-        - batch
         - resourcedetection
         - resource
         {{- if .Values.environment }}
         - resource/add_environment
         {{- end }}
+        - batch
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:707]: Same. No need to change it here. Resource enrichment processors should work better on batches

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 5]
```diff
@@ -732,6 +731,7 @@ service:
         {{- if .Values.isWindows }}
         - metricstransform
         {{- end }}
+        - batch
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:734]: Same here

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 6]
```diff
@@ -751,10 +751,10 @@ service:
       receivers: [prometheus/agent]
       processors:
         - memory_limiter
-        - batch
         - resource/add_agent_k8s
         - resourcedetection
         - resource
+        - batch
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:757]: and here

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 7]
```diff
@@ -186,6 +185,7 @@ service:
         {{- if .Values.environment }}
         - resource/add_environment
         {{- end }}
+        - batch
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:188]: no change needed here as well

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 8]
```diff
@@ -195,14 +195,14 @@ service:
...
         - memory_limiter
-        - batch
         - resource/add_cluster_name
         {{- if .Values.extraAttributes.custom }}
         - resource/add_custom_attrs
         {{- end }}
         {{- if (and .Values.splunkPlatform.metricsEnabled .Values.environment) }}
         - resource/add_environment
         {{- end }}
+        - batch
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:205]: same

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 9]
```diff
@@ -227,9 +227,9 @@ service:
       processors:
         - memory_limiter
         - k8sattributes
-        - batch
         - filter/logs
         - resource/logs
+        - batch
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:232]: Please move it right after `filter/logs`

File: helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl [hunk 10]
```diff
@@ -245,10 +245,10 @@ service:
       receivers: [prometheus/collector]
       processors:
         - memory_limiter
-        - batch
         - resource/add_collector_k8s
         - resourcedetection
         - resource/add_cluster_name
+        - batch
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:251]: No change needed here

File: helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl [hunk 11]
```diff
@@ -188,7 +188,7 @@ service:
     # k8s metrics pipeline
     metrics:
       receivers: [k8s_cluster]
-      processors: [memory_limiter, batch, resource, resource/k8s_cluster]
+      processors: [memory_limiter, resource, resource/k8s_cluster, batch]
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl:191]: Same. No changes needed in this file

**Comment by dmitryax** [comment]
Please revert most of the changes unless you can prove performance improvement with some benchmarks

**Comment by harshit-splunk** [comment]
I have reverted the changes. I have converted this PR to a draft. I am not sure how I can benchmark the pipeline. Do you have any references for how we have done it in past?

**Comment by atoulme** [comment]
You might want to look at https://github.com/signalfx/splunk-otel-collector-chart/tree/main/test to get started, and see if you can add a test that emulates enough load.

File: helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl [hunk 12]
```diff
@@ -634,9 +634,9 @@ service:
         - groupbyattrs/logs
         {{- end }}
         - k8sattributes
-        - batch
         {{- if not $gateway.enabled }}
         - filter/logs
+        - batch
         {{- end }}
```
Comment (dmitryax) [inline helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:640]:
```suggestion
        {{- end }}
        - batch
```

**Comment by dmitryax** [comment]
I mean we need benchmarks if you want to push the batch processor to the very end. I don't really see a reason to do that

**Comment by dmitryax** [comment]
Please rebase

**Review by dmitryax — APPROVED** [review]

**Pull request merged**